### PR TITLE
Add unit test for HandShakeInfo.Equal

### PR DESCRIPTION
**Pull request by janardhanvissa**

### CODE REVIEW

File: internal/credentials/xds/handshake_info_test.go [hunk 1]
```diff
@@ -300,3 +306,120 @@ func TestMatchingSANExists_Success(t *testing.T) {
...
+			}, false),
+			expected: false,
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.desc, func(t *testing.T) {
+			got := test.hi1.Equal(test.hi2)
+			if got != test.expected {
+				t.Errorf("hi1.Equal(hi2) = %v; want %v", got, test.expected)
```
Comment (purnesh42H) [inline internal/credentials/xds/handshake_info_test.go]: got before want

Comment (janardhanvissa) [reply internal/credentials/xds/handshake_info_test.go]: Above comments are addressed and pushed the changes.

File: internal/credentials/xds/handshake_info_test.go [hunk 2]
```diff
@@ -300,3 +312,114 @@ func TestMatchingSANExists_Success(t *testing.T) {
 func newStringP(s string) *string {
 	return &s
 }
+
+func TestHandshakeInfo_Equal(t *testing.T) {
```
Comment (purnesh42H) [inline internal/credentials/xds/handshake_info_test.go:316]: Name can be just `TestEqual`

File: internal/credentials/xds/handshake_info_test.go [hunk 3]
```diff
@@ -300,3 +312,114 @@ func TestMatchingSANExists_Success(t *testing.T) {
...
+			hi2: &HandshakeInfo{
+				rootProvider:      &mockCertProvider{id: 1},
+				identityProvider:  mockProvider1,
+				sanMatchers:       nil,
+				requireClientCert: false,
+			},
+			wantMatch: false,
+		},
+		{
+			desc:      "identical mockCertProvider instances",
```
Comment (purnesh42H) [inline internal/credentials/xds/handshake_info_test.go:399]: looks like this case is repeated?

**Review by purnesh42H — APPROVED** [review]
lgtm. Feel free to open on main repo

**Comment by janardhanvissa** [comment]
Re-opening the PR

**Pull request closed without merging**